pkg/uroot/uroot.go

```diff
@@ -140,21 +140,28 @@ func CreateInitramfs(opts Opts) error {
 
 	// Add files from command line.
 	for _, file := range opts.ExtraFiles {
+		var src, dst string
 		parts := strings.SplitN(file, ":", 2)
-		path, err := filepath.Abs(filepath.Join(parts...))
-		if len(parts) != 2 {
-			parts = append(parts, parts[0][1:])
+		if len(parts) == 2 {
+			// treat the entry with the new src:dst syntax
+			src = parts[0]
+			dst = parts[1]
+		} else {
+			// plain old syntax
+			src = file
+			dst = file
 		}
+		src, err := filepath.Abs(src)
 		if err != nil {
-			return fmt.Errorf("couldn't find absolute path for %q: %v", file, err)
+			return fmt.Errorf("couldn't find absolute path for %q: %v", src, err)
 		}
-		if err := archive.AddFile(path, parts[1]); err != nil {
+		if err := archive.AddFile(src, dst); err != nil {
```

Comment:
I think you are misinterpreting the way the syntax is used; but your way is a little more general.

Intention of the existing code is something like this: "/usr/local/go:bin/go" I.e. src is /usr/local/go/bin/go and dst is bin/go.

@rminnich opinion on this?

Reply:
Oooooh. I'm sorry. I totally didn't get this.

Reply:
okay, it seems like we're both right here, but i'm just too tired to get this completely right now. i'll look at this tomorrow again...

Reply:
I pretty much have no objection to this change, it allows more flexible path rewriting, actually:
/my/etchosts:/etc/hosts
/mybin:/bin

Just be aware that it's pretty important that the dst NOT BE ALLOWED to begin with a / or sooner or later someone will trash their local root file system with a carelessly typed sudo. so if 
fllepath.IsAbs(filepath.Clean(dst)) is true then dst should be fixed up. Can you add that check?